amqp.go

```diff
@@ -175,6 +180,31 @@ type TimeLimitRequest struct {
 	InvocationID string
 }
 
+// EmailRequest defines the structure of a request to be sent to iplant-email.
+type EmailRequest struct {
+	TemplateName        string                 `json:"template"`
+	TemplateValues      map[string]interface{} `json:"values"`
+	Subject             string                 `json:"subject"`
+	ToAddress           string                 `json:"to"`
+	CourtesyCopyAddress string                 `json:"cc,omitempty"`
+	FromAddress         string                 `json:"from-addr,omitempty"`
+	FromName            string                 `json:"from-name,omitempty"`
+}
+
+// NotificationMessage defines the structure of a notification message sent to
+// the Discovery Environment UI.
+type NotificationMessage struct {
+	Deleted       bool                   `json:"deleted"`
+	Email         bool                   `json:"email"`
+	EmailTemplate string                 `json:"email_template"`
+	Message       map[string]interface{} `json:"message"`
+	Payload       interface{}            `json:"payload"`
```

Comment:
The legacy handler in the `event-recorder` service needs to be able to store a struct in the `Payload` field, so it uses interface. For the time being, the `Message` field always contains a map, so it's defined as a map in here for convenience. With the field defined as a map, we can examine its contents without having to do a type switch or a type assertion.